worktree.go

```diff
@@ -663,14 +663,20 @@ func (w *Worktree) checkoutChangeSubmodule(name string,
 ) error {
 	switch a {
 	case merkletrie.Modify:
-		sub, err := w.Submodule(name)
+		subs, err := w.Submodules()
 		if err != nil {
 			return err
 		}
-
-		if !sub.initialized {
-			return nil
+		for _, sub := range subs {
+			if sub.Config().Path == name {
+				if !sub.initialized {
+					return nil
+				}
+				goto found
```

Comment:
Why `goto` rather than just `return` here?

Reply:
I'm sorry that the exhaustion from debugging until late at night made me write such ugly code. I've fixed it.